FWCore/Framework/bin/cmsRun.cpp

```diff
@@ -298,8 +298,8 @@ int main(int argc, char* argv[]) {
       oneapi::tbb::task_arena arena(nThreads);
       arena.execute([&]() {
         context = "Constructing the EventProcessor";
-        EventProcessorWithSentry procTmp(
-            std::make_unique<edm::EventProcessor>(processDesc, jobReportToken, edm::serviceregistry::kTokenOverrides));
+        EventProcessorWithSentry procTmp(std::make_unique<edm::EventProcessor>(
+            std::move(processDesc), jobReportToken, edm::serviceregistry::kTokenOverrides));
```

Comment:
During beginJob, EventSetupsController is using pointers into the ProcessDesc ParameterSet while it is dealing with sharing of ESProducers amongst the top level process and SubProcesses. When EventSetupsController is done with this, it explicitly deletes the objects holding the pointers to ParameterSets before completion of beginJob. The easiest fix to this would be to replace the move with an explicit
```
processDesc.reset()
```
 5 lines lower after the call to beginJob. Given that a lot of the memory pressure hits at the first beginRun maybe that is good enough. Or I could start digging into the EventSetupsController code and try to move that work earlier. I don't know how difficult that would be yet.

Reply:
> The easiest fix to this would be to replace the move with an explicit
> 
> ```
> processDesc.reset()
> ```
> 
> 5 lines lower after the call to beginJob.

Thanks David for the suggestion, I just did that.